py/objarray.c

```diff
@@ -405,12 +415,17 @@ STATIC mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value
                     mp_seq_replace_slice_grow_inplace(o->items, o->len,
                         slice.start, slice.stop, src_items, src_len, len_adj, item_sz);
                 } else {
-                    mp_seq_replace_slice_no_grow(o->items, o->len,
-                        slice.start, slice.stop, src_items, src_len, item_sz);
-                    // Clear "freed" elements at the end of list
-                    // TODO: This is actually only needed for typecode=='O'
-                    mp_seq_clear(o->items, o->len + len_adj, o->len, item_sz);
-                    // TODO: alloc policy after shrinking
+                    if (o->base.type == &mp_type_memoryview) {
+                        // destination is memoryview use memcpy() instead of mp_seq_replace_slice_no_grow() below
+                        memcpy((uint8_t*)o->items + o->free + slice.start, src_items, src_len);
```

Comment:
mp_seq_replace_slice_no_grow() calls memcpy() and memmove(), I wanted to avoid calling memmove() even if it results in a no-op. However, the code above does not handle correctly arrays with item-size != 1. When dealing with a memory view rhs I *think* I need to call mp_seq_replace_slice_no_grow() as:
```
src_len = src_slice->len;
src_items = (uint8_t*)src_slice->items + (src_slice->free * item_sz);
...
mp_seq_replace_slice_no_grow((uint8_t*)o->items + (o->free * item_sz), o->len, slice.start, slice.stop, src_items, src_len, item_sz);
```
or replace it with an equivalent memcpy().